crates/cheatcodes/src/inspector.rs

```diff
@@ -1569,6 +1598,44 @@ impl Cheatcodes {
         }
     }
 
+    #[cold]
+    fn meter_gas_record<DB: DatabaseExt>(
+        &mut self,
+        interpreter: &mut Interpreter,
+        ecx: &mut EvmContext<DB>,
+    ) {
+        if matches!(interpreter.instruction_result, InstructionResult::Continue) {
+            self.gas_metering.gas_records.iter_mut().for_each(|record| {
+                if ecx.journaled_state.depth() == record.depth {
+                    // Skip the first opcode of the first call frame as it includes the gas cost of
+                    // creating the snapshot.
+                    if self.gas_metering.last_gas_used != 0 &&
+                    // Skip updating of gas recording for opcodes that create new call frames.
+                    !matches!(
+                            interpreter.current_opcode(),
```

Comment:
The proposed change makes sense to me, it is more clear that way. cc @klkvr would you agree?

Reply:
hmm, I am unsure that this branch is correct in general. why would we want to skip accounting for those opcodes gas? in earlier iteration we've accounted for it in `call_end`/`create_end` hooks, but looks like this would just miss calls gas usage data?

Reply:
Fixed, previously it was necessary but not anymore with this implementation. The skipping over the first opcode is still necessary as it includes gas overhead that is unreliable and context-specific.

```
{
  "testGasComparisonCreateA": "99168",
  "testGasComparisonCreateB": "99168",
  "testGasComparisonEmptyA": "0",
  "testGasComparisonEmptyB": "0",
  "testGasComparisonExternalA": "26776",
  "testGasComparisonExternalB": "26776",
  "testGasComparisonFlareA": "5821936",
  "testGasComparisonFlareB": "5821936",
  "testGasComparisonInternalColdA": "22100",
  "testGasComparisonInternalColdB": "22106",
  "testGasComparisonInternalWarmA": "100",
  "testGasComparisonInternalWarmB": "106",
  "testGasComparisonNestedCallsA": "31955",
  "testGasComparisonNestedCallsB": "31955"
}
```